buildSrc/subprojects/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```diff
@@ -247,20 +246,6 @@ fun removeTeamcityTempProperty() {
 val Project.maxParallelForks: Int
     get() = findProperty("maxParallelForks")?.toString()?.toInt() ?: 4
 
-fun failOnEmptySourceFolder() {
-    // Empty source dirs produce cache misses, and are not caught by `git status`.
-    // Fail if we find any.
-    tasks.withType<SourceTask>().configureEach {
-        doFirst {
-            source.visit {
-                if (file.isDirectory && file.listFiles()?.isEmpty() == true) {
-                    throw IllegalStateException("Empty src dir found. This causes build cache misses. See github.com/gradle/gradle/issues/2463.\nRun the following command to fix it.\nrmdir ${file.absolutePath}")
```

Comment:
Instead of deleting the whole code I'd actually be interested in issuing warnings that they are empty directories. WDYT?

Reply:
Why would we care that they still exist though? And how should we emit the warnings?
I'm not sure about performance, but if we want to keep the code there, that would mean that we would still have to visit every file in the source tree in the `doFirst` action.